### PR TITLE
Test: replace the bundled msgpuck library with a FetchContent call

### LINKED ISSUE #60: Download MsgPuck library during test build
We just copied MsgPuck library to the test directory ([here](https://github.com/tarantool/tntcxx/blob/master/test/msgpuck.h) and [here](https://github.com/tarantool/tntcxx/blob/master/test/hints.c)). Let's remove it from our tree and download it with CMake `FetchContent` during test build.

### PR DESCRIPTION
**Pull request by CuriousGeorgiy**
This patch replaces the bundled msgpuck library with a FetchContent call

Closes #60

### CODE REVIEW

File: test/EncDecGPerfTest.cpp [hunk 1]
```diff
@@ -36,7 +36,6 @@
 #include <random>
 
 #include "../src/mpp/mpp.hpp"
-#define MP_LIBRARY 1
```
Comment (drewdzzz) [inline test/EncDecGPerfTest.cpp:39]: Why did you remove this define?

**Review by alyapunov — APPROVED** [review]
